packages/validator/src/validator.ts

```diff
@@ -20,12 +19,31 @@ import {ChainHeaderTracker} from "./services/chainHeaderTracker";
 import {MetaDataRepository} from ".";
 import {toHexString} from "@chainsafe/ssz";
 import {ValidatorEventEmitter} from "./services/emitter";
+import {ValidatorStore} from "./services/validatorStore";
+
+export enum SignerType {
+  Local,
+  Remote,
+}
+
+export type Signers =
+  | {
+      type: SignerType.Local;
+      secretKeys: SecretKey[];
+    }
+  | {
+      type: SignerType.Remote;
+      url: string;
+      pubkeys: PublicKey[];
+      secretKey: SecretKey;
```

Comment:
You should remove this `secretKey` entirely

Reply:
It was an intentional decision to keep a secretKey attribute inside a signers object of remote type. It was done to maintain consistency with the validators mapping in ValidatorStore (https://github.com/novon-labs/lodestar/blob/novon/validator-remote/packages/validator/src/services/validatorStore.ts#L41). This secretKey's value remains undefined and only helps in making sure a valid mapping of the type Map<PubkeyHex, BLSKeypair> exists in either case: local or remote. This secretKey is initialized with an empty constructor inside CLI (https://github.com/novon-labs/lodestar/blob/novon/validator-remote/packages/cli/src/cmds/dev/handler.ts#L163) and this is done since we need to init() if we want to create a secretKey and we are only doing init in higher level packages like the CLI.

The other option was to use another state variable in ValidatorStore when signingMode=remote, or have the validators state variable accept multiple types. I did opt for the latter option initially but I had to change all the functions in the ValidatorStore class to support validator either having different types or being undefined and I just thought that was unncessary overhead.  

Let me know what you think is the better approach?